src/server/campaignd/server.cpp

```diff
@@ -668,29 +753,111 @@ void server::handle_request_campaign_list(const server::request& req)
 
 void server::handle_request_campaign(const server::request& req)
 {
-	config& campaign = get_campaign(req.cfg["name"]);
+	config& campaign = get_addon(req.cfg["name"]);
 
 	if(!campaign || campaign["hidden"].to_bool()) {
 		send_error("Add-on '" + req.cfg["name"].str() + "' not found.", req.sock);
 		return;
 	}
-	
-	const int size = filesystem::file_size(campaign["filename"]);
 
-	if(size < 0) {
-		send_error("Add-on '" + req.cfg["name"].str() + "' could not be read by the server.", req.sock);
-		return;
+	// The desired version is selected on the client side considering
+	// the min_wesnoth_version in [version] children of the addon info (#TODO: gfgtdf's part)
+	std::string to = req.cfg["version"].str();
+	const std::string& from = req.cfg["from_version"].str();
+	std::string full_pack = campaign["filename"].str();
+
+	auto version_map = get_version_map(campaign);
+
+	if(version_map.empty()) {
+		// Old format ?
```

Comment:
If this is not supposed to happen then the server needs to abort this operation with an error.

Reply:
Ok, added the error message.